### DISCUSSION TITLE
让简悦变成你的知识库（离线且具有永久链接）

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2221] [服务]
# 重要提示

> ### 嗨，简悦全新的 [教程中心](https://www.yuque.com/kenshin/simpread) 已上线，**下面的内容并不一定是最新的**，建议 [访问这里](https://www.yuque.com/kenshin/simpread/wkswh7) 查看最新版教程。

***

> 当你使用 Pocket 类产品的一个很重要原因：它可以缓存稍后读的页面，也就是当原页面不存在时，在 Pocket 里面仍可以看到。

> [如何让稍后读支持读取本地缓存](https://github.com/Kenshin/simpread/discussions/2146)  其实就是对知识库的一种应用方式。

## 什么是知识库

> 简单点说，将你需要的页面，永久保存下来，且辅以不同的工具可以方便检索出这些内容。

> 听起来有点像「第二大脑」？原则上来说它们是一样的，但简悦的知识库在这方面更具优势。

## 优势

> 相比 **印象笔记** 之类的产品，简悦的优势在于：

1. **本地数据**，也就是说，通过这种方式得到的数据是在你的本地，你可以通过 **坚果云 或 Dropbox 同步这些数据到任意设备上**
2. 同时还支持类似 DEVONThinks 这样的检索工具辅助检索。
3. 通用格式，如： `.html` `.md` 文件等。

## 前提

1. **[简悦 2.2.0](http://ksria.com/simpread/welcome/version_2.2.0.html)**
2. **[简悦 · 同步助手 1.0.1](http://ksria.com/simpread/docs/#/Sync?id=%e4%b8%8b%e8%bd%bd)**

## 使用场景

1. 当加入稍后读后，会自动将当前页面下载为 HTML
2. 当标注时，会自动将当前页面下载为 HTML
3. 导出时，会同时导出当前页面的 标注 & 全文（含标注）
4. 双链笔记使用场景，每个标注都可以追述到原文，且是在本地永久保存的原文对应的位置。

## 配置

- [如何让稍后读支持读取本地缓存](https://github.com/Kenshin/simpread/discussions/2146)
- [当产生标注时，如何自动同步到本地](https://github.com/Kenshin/simpread/discussions/2220)
- [如何配置永久链接（在双链笔记中做源链接回溯）](https://github.com/Kenshin/simpread/discussions/3188)

## 如何使用

> 在运行同步助手的前提下，如你的 `output` 有如下文件：

![image](https://user-images.githubusercontent.com/81074/119218290-3bae0d80-bb12-11eb-8817-a34808fa36b2.png)

> 在浏览器输入 `http://localhost:7026/reading/1600` 即可看到类似下图的效果

![image](https://user-images.githubusercontent.com/81074/119218339-5ed8bd00-bb12-11eb-9982-9188cbd1f9a8.png)

> 为什么会这样？同步助手在本地开了一个服务环境，这样就能通过  `http://localhost:7026/reading/xxx` 的方式打开对应的文件。 

> 在双链笔记中做永久链接，并回溯到原文中

https://user-images.githubusercontent.com/81074/145700385-bb58cf64-fd10-4a6e-8a60-de846b1a3257.mp4

## 注意

> 上图中我只输入了 `http://localhost:7026/reading/1600` 而并不是全名 e.g.  `http://localhost:7026/reading/1600-xxxx` ，原因在于：

> 同步助手会先获取 `id` ，而后才是 `标题`，因此下面的 URL 都指向了一个

- `http://localhost:7026/reading/1600`
- `http://localhost:7026/reading/1600-为了完美阅读模式这个小目标 ，我适配了 160+ 个网站，因此诞生了它 - 简悦 ：沉浸式阅读的 Chrome 扩展`

> 当不存在 `id-` 时，则通过 `文件名` 得到结果

## 本地全文搜索

> 使用 DEVONThink 或 File Locator 之类的进行文本内搜索。

![2021-05-22_15-37-26 (1)](https://user-images.githubusercontent.com/81074/119218605-ce02e100-bb13-11eb-87c3-b5fbb9bb88d8.gif)

## 引申

1. 不止可以在本地使用，也可以在线查看，详细请看 [资料库](https://github.com/Kenshin/simpread/discussions/2956)。
2. 通过定制化 Markdown，可方便的支持任意双链笔记，详细请看 [通过定制化导出功能支持导入到任意双链笔记](https://github.com/Kenshin/simpread/discussions/2222)。

## 参考知识点

> 上述操作都已经写到了文档里面，有感兴趣的用户，可以看下。

- 简悦 · 同步助手的 [增强导出功能](http://ksria.com/simpread/docs/#/Sync?id=%e5%af%bc%e5%87%ba%e6%9c%8d%e5%8a%a1)
- [自定义标题](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=%e8%87%aa%e5%ae%9a%e4%b9%89%e5%af%bc%e5%87%ba%e6%a0%87%e9%a2%98)
- [自动化](http://ksria.com/simpread/docs/#/%E8%87%AA%E5%8A%A8%E5%8C%96)

**Comment by GreenHatHG** [comment]
请问下这个保存的英文文章怎么加上双语翻译呢
主要想法是这样，网页保存到稍后读，电脑端浏览器阅读，阅读时候双语翻译，批注，笔记同步到双链笔记，下次从双链笔记打开永久链接时候就是之前已经翻译好带笔记的网页

**Reply by Kenshin** [reply]
嗨，我应该在 [简悦的 Telegram 群](https://t.me/simpreadgroup) 回复你了，再回复一下：

1. 将需要使用彩云的页面加入延迟加载列表
2. 在原页面上使用双语
3. 进入阅读模式（而后就有双语了）
4. 标注（会自动加稍后读）

**Reply by Kenshin** [reply]
另外，你可以提供个 URL，我可以做成教程，方便更多用户。

**Comment by rfiyGH** [comment]
所有设置都成功了，但是稍后阅读里面的每一篇文章都加载不出来，显示“小憩片刻，马上就好”，怎么解决呢？

**Reply by Kenshin** [reply]
1. 请比对你的配置与 [教程](https://kb.simpread.pro/#/page/建立知识库)  区别

2. 本地需要有 `idx-xxx.html` 文件，且在 `output` 文件夹

3. 正确配置后，应该出现下图效果

   ![image](https://user-images.githubusercontent.com/81074/172805520-26cc8275-a70d-4743-993e-1372cf8f3585.png)

4. 访问 http://localhost:7026/reading/index 应该有值（说明步骤 2 是正确的）

5. 留意下面的选项是否开了

   ![image](https://user-images.githubusercontent.com/81074/172805722-d04bac2c-b578-4ec5-8137-689abfca0ab7.png)

**Reply by rfiyGH** [reply]
我检查了下，你说的1、2、3、4我设置的是正确的，5的选项也都开启了。

**Reply by Kenshin** [reply]
> 这篇则是通过raindrop保存的，它能同步到稍后阅读界面，但是加载不出。会一直显示“小憩片刻，马上就好”

这么说的话，这是算「正常」的。

通过 API 导入的页面不是 100% 都能显示，你可以试试开启内置解析，如果还不行的话，只能通过打开原页面后进入阅读模式的方式重新生成 HTML。

API 的快照功能跟这个无关，API 的快照不可能 100% 都完美抓取，你可以理解是个「爬虫」，帮你爬取页面而已。

**Reply by andyboyce** [reply]
里面加载的图片还是网络的，一般博客或者网站挂了，图片也会挂掉，可以直接让图片也本地话吗？或者有能让图片自动保存到自己的服务里面然后html里面加载自己服务的图片？

**Reply by Kenshin** [reply]
@andyboyce 

是的，所以在同步打开原文 → 阅读模式 → 导出快照（离线 HTML）的方式即可解决。

注意：前提是通过 API 导入

**Reply by andyboyce** [reply]
@Kenshin 是增强导出里面设置吗，我目前的是html, 如果选择离线HTML，有如下这个提示：
使用 离线 HTML 时，会将图片资源打包到 HTML 中，而这种转换比较消耗性能，所以每产生一个标注（增删改）都会做一次这样的操作，因此 不建议在标注时选择离线 HTML or markdown。 ps: 目前还未熟练使用，有蛮多不太明白的地方，这里说道性能问题，是因为每次修改一个地方的标注，会同步整个所有的文章，还是仅当前文章呢？--所有的数据都在一个配置文件里面，猜测是同步了所有的文章[数据]是吧，所以说会有性能问题？

**Reply by Kenshin** [reply]
> 说道性能问题，是因为每次修改一个地方的标注，会同步整个所有的文章，还是仅当前文章呢？

仅当前文章，但如果是文章的图片较多，且都是以 `base64` （离线方式）保存的话，每保存一次就会转换一次，这是这个提示出现的原因。

**Comment by freeey** [comment]
导出完整markdown的时候提示：“离线下载的文件体积较大，所以需要使用Chrome下载方案，请授权。”怎么解决啊？

**Reply by Kenshin** [reply]
不建议使用 **Markdown 离线版**，可以试试 [Textbundle](https://simpread.pro/docs/#/Textbundle) 或 [md + assets](https://github.com/Kenshin/simpread/discussions/4041) 或者 [离线 HTML](https://simpread.pro/docs/#/%E7%A6%BB%E7%BA%BFHTML) 都可以。

**Reply by Alice-Shi** [reply]
> [离线 HTML](https://simpread.pro/docs/#/%E7%A6%BB%E7%BA%BFHTML) 都可以

刚刚尝试离线下载这篇文章，不论是 离线html 还是 离线 markdown 👉 https://mp.weixin.qq.com/s/J8Qa3IRqrhFaGZzSNrNRKA
都是同样报错`“离线下载的文件体积较大，所以需要使用Chrome下载方案，请授权。`

**Reply by Kenshin** [reply]
@Alice-Shi 

我测试了下，是没问题的。建议在**隐身模式下仅使用简悦的方式**，排除其它扩展造成的干扰。

如果仍有问题请通过 https://github.com/kenshin/simpread/issues/new 讨论。

**Reply by Alice-Shi** [reply]
> 建议在**隐身模式下仅使用简悦的方式**，排除其它扩展造成的干扰。

请问指的是 `New Incognito Window 无痕浏览` 吗？在这个模式下 所有拓展 都无效呢。

**Reply by Kenshin** [reply]
@Alice-Shi 

<img width="702" alt="image" src="https://user-images.githubusercontent.com/81074/219830362-40e074f2-4301-4c16-ba05-88b5db588488.png">

**Comment by xuxingll** [comment]
方便的话可以弄个PC的设置图文，因为PC的环境可能各种不一样。比如我在同步助手里设置增强导出路径，两台电脑“基本一样”的设置，结果预期还不一样。笔记本上面路径似乎怎么都不生效，不知道哪里问题。我的需求是obsidian联动里只保存annote，assets导出文件到其他目录。

**Reply by Kenshin** [reply]
这个教程的最前面就有你需要的教程链接 https://www.yuque.com/kenshin/simpread/wkswh7

另外，如果是新用户的话，还可以直接使用配置库方案 https://www.yuque.com/kenshin/simpread/ds8zk0

Obsidian 配置库 https://www.yuque.com/kenshin/simpread/psugef
